py/objfun.h

```diff
@@ -28,11 +28,19 @@
 
 #include "py/obj.h"
 
+#if MICROPY_PY_SYS_TRACE
+#include "py/emitglue.h"
+#endif
+
 typedef struct _mp_obj_fun_bc_t {
     mp_obj_base_t base;
     mp_obj_dict_t *globals;         // the context within which this function was defined
     const byte *bytecode;           // bytecode for the function
     const mp_uint_t *const_table;   // constant table
+#if MICROPY_PY_SYS_TRACE
+    const mp_raw_code_t *rc;
+    mp_obj_t line_of_definition;
```

Comment:
This is clearly superfluous, line info is already stored with bytecode object.

Reply:
In the case a function was NOT defined at compile time this variable is needed; e.g. *eval*.

Reply:
From what I understand the bytecode contains the line info of the first opcode of the first expression INSIDE the block. Not the line where the block was created; e.g. *MP_BC_MAKE_FUNCTION*.

```Python
283:    def do(): # MP_BC_MAKE_FUNCTION Somehwere inside the bytecode `AAA`
284:        
285:        print("Who loves the sun?") # bytecode `BBB` starts here
...
301:    do()
```

So when there is a bytecode execution switch from `AAA` to `BBB` I need the line number of the **def** in order to trigger the *call* trace event with the correct line numbering.

Trace event *call* is triggered at line 301 and jumps to line 285 but the called function is defined at line 283. This information needs to be somehow propagated. Hence the `line_of_definition`. If you have another method please tell me.

Reply:
> From what I understand the bytecode contains the line info of the first opcode of the first expression INSIDE the block.

Ok, if the difference is important for your usecase, please add code comment explaining this.